crates/cosmos/cosmos-chain-components/src/impls/transaction/convert_gas_to_fee.rs

```diff
@@ -0,0 +1,132 @@
+use core::cmp::min;
+
+use cgp::prelude::CanRaiseError;
+use hermes_chain_type_components::traits::fields::chain_id::HasChainId;
+use hermes_relayer_components::transaction::traits::types::fee::HasFeeType;
+use ibc_proto::cosmos::base::v1beta1::Coin;
+use ibc_proto::cosmos::tx::v1beta1::Fee;
+use ibc_relayer::chain::cosmos::gas::{mul_ceil, mul_floor};
+use ibc_relayer::error::Error as RelayerError;
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
+
+use crate::traits::convert_gas_to_fee::GasToFeeConverter;
+use crate::traits::eip::eip_query::CanQueryEipBaseFee;
+use crate::traits::gas_config::HasGasConfig;
+use crate::traits::rpc_client::HasRpcClient;
+
+pub struct StaticConvertCosmosGasToFee;
+
+impl<Chain> GasToFeeConverter<Chain> for StaticConvertCosmosGasToFee
+where
+    Chain: HasFeeType<Fee = Fee>
+        + HasChainId<ChainId = ChainId>
+        + HasRpcClient
+        + HasGasConfig
+        + CanRaiseError<&'static str>,
+{
+    async fn gas_amount_to_fee(chain: &Chain, gas_used: u64) -> Result<Chain::Fee, Chain::Error> {
+        let gas_config = chain.gas_config();
+        let adjusted_gas_limit =
+            adjust_estimated_gas(gas_config.gas_multiplier, gas_config.max_gas, gas_used);
+
+        // The fee in coins based on gas amount
+        let fee_amount = mul_ceil(adjusted_gas_limit, gas_config.gas_price.price);
+
+        let amount = Coin {
+            denom: gas_config.gas_price.denom.to_string(),
+            amount: fee_amount.to_string(),
+        };
+
+        Ok(Fee {
+            amount: vec![amount],
+            gas_limit: adjusted_gas_limit,
+            payer: "".to_string(),
+            granter: gas_config.fee_granter.clone(),
+        })
+    }
+}
+
+pub struct DynamicConvertCosmosGasToFee;
+
+impl<Chain> GasToFeeConverter<Chain> for DynamicConvertCosmosGasToFee
+where
+    Chain: HasFeeType<Fee = Fee>
+        + HasChainId<ChainId = ChainId>
+        + HasRpcClient
```

Comment:
Constraints that are only required by `StaticConvertCosmosGasToFee` can be removed, as we already have the constraint `StaticConvertCosmosGasToFee: GasToFeeConverter<Chain>`.

Reply:
Done in baace016a4e0d4040d04b7ddceac5baa0244c9d6